upload_components/upload.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+IFS=';' read -ra DIRECTORIES <<<"${COMPONENTS_DIRECTORIES:-.}"
+NAMESPACE=${COMPONENTS_NAMESPACE:-espressif}
+
+NUMBER_OF_DIRECTORIES="${#DIRECTORIES[@]}"
+echo "Processing $NUMBER_OF_DIRECTORIES components"
+
+for ITEM in "${DIRECTORIES[@]}"; do
+    FULL_PATH="${GITHUB_WORKSPACE?}/${ITEM}"
+    if [ "$NUMBER_OF_DIRECTORIES" -eq "1" ]; then
+        NAME=${COMPONENT_NAME?}
+    else
+        NAME=$(basename "$(realpath "${FULL_PATH}")")
+    fi
+
+    echo "Processing component \"$NAME\" at $ITEM"
+    python3 -m idf_component_manager upload-component --path="${FULL_PATH}" --namespace="${NAMESPACE}" --name="${NAME}"
+    EXIT_CODE=$?
+    if [ "$EXIT_CODE" -ne "0" ] && [ "$EXIT_CODE" -ne "144" ]; then
```

Comment:
What is the meaning of code `144`?

Reply:
This code is returned when the version is already on the service. 

144 refers to x86 NOP `0x90`